### DISCUSSION TITLE
`perSystem.nci.outputs.default`

### DISCUSSION THREAD

**Discussion by mightyiam** [90-008/nix-cargo-integration#163] [Q&A] [answered]
`perSystem.nci.outputs` seems to have an attribute `default` that is an empty set. Is there a purpose to this, please?

**Comment by 90-008** [comment]
i don't see this anywhere, so i'm not sure? where exactly are you seeing this? `nci.outputs` is just a read-only option putting the generated packages, devshells etc. in, it would not have a `default` unless a crate or project was named so (and if it was it would not be empty).

**Reply by mightyiam** [reply]
Sorry, I do have a project named `default`! So... I guess that explains the existence of the attribute. What about its emptiness? Is there something wrong with my project?

**Reply by 90-008** [reply]
it should not be empty, that is weird. i'll look into it, can you also link your project if possible? or a reproducing example at least.

**Reply by mightyiam** [reply]
This. Practically a stash branch. Sorry.
https://github.com/mobusoperandi/website/tree/mob/nix-build

**Reply by 90-008** [reply]
it seems to be working fine for me, i'm not sure what's wrong? it looks like it exports properly with `nix flake show`, i can see the packages and all. setting `debug = true` in `mkFlake` and looking at `debug.allSystems.x86_64-linux.nci.outputs.default` also seems fine, it has the devshell defined. note that nothing other than the devshell will get defined for the project, crates are outputted separately under outputs, if that's what you were confused about?

**Reply by mightyiam** [reply, marked as answer]
Inspected again and it's non-empty :shrug:
